File: server/Data/Dtos/Shared/OngoingPosition/Request/OngoingPositionRequestDto.cs
using System;
using System.ComponentModel.DataAnnotations;
using SolidTradeServer.Data.Models.Annotations;
using SolidTradeServer.Data.Models.Enums;
using static SolidTradeServer.Common.ErrorMessages;

namespace SolidTradeServer.Data.Dtos.Shared.OngoingPosition.Request
{
    public class OngoingPositionRequestDto
    {
        [Required] 
        public string Isin { get; set; }

        [Required]
        [Range(0.00010, int.MaxValue, ErrorMessage = PriceThresholdMessage)]
        public decimal PriceThreshold { get; set; }

        [Required]
        public EnterOrExitPositionType? Type { get; set; }
        
        [Required]
        [IsFutureDate(ErrorMessage = GoodUntilErrorMessage)]
        public DateTimeOffset? GoodUntil { get; set; }
        
        [Range(1, int.MaxValue)]
        public decimal NumberOfShares { get; set; }
    }
}

File: client/lib/data/dtos/shared/request/ongoing_position_request_dto.dart
import 'package:dart_json_mapper/dart_json_mapper.dart';
import 'package:solidtrade/data/models/enums/entity_enums/enter_or_exit_position_type.dart';

@jsonSerializable
class OngoingPositionRequestDto {
  final String isin;
  final DateTime goodUntil;
  final double priceThreshold;
  final double numberOfShares;
  final EnterOrExitPositionType type;

  OngoingPositionRequestDto({
    required this.isin,
    required this.goodUntil,
    required this.priceThreshold,
    required this.numberOfShares,
    required this.type,
  });
}
